src/scgpt/embedding/script.py

```diff
@@ -145,9 +145,23 @@ def setup_logger():
     pre_norm=False  #TODO: Parametrize when GPU-based machine types are supported
     )
 
+
+logger.info("Loading model")
+model_file = par["model"]
+model_dict = torch.load(model_file, map_location=device)
+
+# Ensure the provided model has the correct architecture
+finetuned_checkpoints_key = par.get("finetuned_checkpoints_key", None)
+if par["finetuned_checkpoints_key"]:
```

Comment:
This will fail if the argument is None. I guess this is a refactoring error

```suggestion
finetuned_checkpoints_key = par.get("finetuned_checkpoints_key", None)
if finetuned_checkpoints_key:
```